src/test/unit/parameter_group.ld

```diff
@@ -0,0 +1,10 @@
+SECTIONS {
+  .pg_registry :   SUBALIGN(4)  /* force 4-byte alignment for pg_regisry. Gcc default is 32 bytes, causing padding insertion and breaking the init structure. */
+  {
+    PROVIDE_HIDDEN (__pg_registry_start = .);
+    KEEP (*(.pg_registry))
+    KEEP (*(SORT(.pg_registry.*)))
+    PROVIDE_HIDDEN (__pg_registry_end = .);
+  }
+}
+INSERT AFTER .fini_array;
```

Comment:
here's an except when using `INSERT BEFORE .data`

```
 .text.unlikely
                0x00432ee8        0x0 /usr/lib/gcc/i686-pc-cygwin/4.9.2/crtend.o
 *fill*         0x00432ee8        0x8 
 .text.startup  0x00432ef0       0x10 /usr/lib/gcc/i686-pc-cygwin/4.9.2/crtend.o
 *(.gnu.linkonce.t.*)
 *(.glue_7t)
 *(.glue_7)
                0x00432f00                ___CTOR_LIST__ = .
                0x00432f00                __CTOR_LIST__ = .
                0x00432f00        0x4 LONG 0xffffffff
 *(.ctors)
 .ctors         0x00432f04        0x4 ../../obj/test/serial_msp_unittest.o
 .ctors         0x00432f08        0x4 ../../obj/test/gtest_main.a(gtest-all.o)
 .ctors         0x00432f0c        0x4 ../../obj/test/gtest_main.a(gtest_main.o)
 *(.ctor)
 *(SORT(.ctors.*))
 .ctors.65535   0x00432f10        0x4 /usr/lib/gcc/i686-pc-cygwin/4.9.2/crtend.o
                0x00432f14        0x4 LONG 0x0
                0x00432f18                ___DTOR_LIST__ = .
                0x00432f18                __DTOR_LIST__ = .
                0x00432f18        0x4 LONG 0xffffffff
 *(.dtors)
 *(.dtor)
 *(SORT(.dtors.*))
                0x00432f1c        0x4 LONG 0x0
 *(.fini)
 *(.gcc_exc)
                0x00432f20                PROVIDE (etext, .)
                0x00432f20                PROVIDE (_etext, .)
 *(.gcc_except_table)
 .gcc_except_table
                0x00432f20      0xa00 ../../obj/test/serial_msp_unittest.o
 .gcc_except_table
                0x00433920     0x1f88 ../../obj/test/gtest_main.a(gtest-all.o)
LOAD /usr/lib/gcc/i686-pc-cygwin/4.9.2/../../../crt0.o
LOAD /usr/lib/gcc/i686-pc-cygwin/4.9.2/crtbegin.o

.pg_registry    0x004358a8      0x218
                0x004358a8                PROVIDE (__pg_registry_start, .)
 *(.pg_registry)
 .pg_registry   0x004358a8       0x20 ../../obj/test/serial_msp_unittest.o
 *(SORT(.pg_registry.*))
                0x004358c8                PROVIDE (__pg_registry_end, .)

.data           0x00436000      0x200
                0x00436000                __data_start__ = .
 *(.data)
```

I'm by no-means an LD expert here so some assistance would be great.